agent-inject/agent/annotations.go

```diff
@@ -173,6 +179,13 @@ func Init(pod *corev1.Pod, image, address, authPath, namespace string) error {
 		pod.ObjectMeta.Annotations[AnnotationAgentRequestsMem] = DefaultResourceRequestMem
 	}
 
+	if _, ok := pod.ObjectMeta.Annotations[AnnotationPreserveSecretCase]; !ok {
+		pod.ObjectMeta.Annotations[AnnotationPreserveSecretCase] = "false"
```

Comment:
This should be a bool to be consistent with other annotations: https://github.com/hashicorp/vault-k8s/blob/master/agent-inject/agent/annotations.go#L210-L244